src/shared-data.js

```diff
@@ -0,0 +1,91 @@
+// Initial state
+const internalSharedData = {
+  openInEditorHost: '/',
+  classifyComponents: true
+}
+
+// ---- INTERNALS ---- //
+
+let Vue
+let bridge
+// Storage API
+let storage = null
+// List of fields to persist to storage (disabled if 'false')
+// This should be unique to each shared data client to prevent conflicts
+let persist = false
+// For reactivity, we wrap the data in a Vue instance
+let vm
+
+export function init (params) {
+  // Mandatory params
+  bridge = params.bridge
+  Vue = params.Vue
+
+  if (params.hasOwnProperty('persist')) {
+    persist = params.persist
+
+    if (!params.hasOwnProperty('storage')) {
+      throw new Error('Missing `storage` params.')
+    }
+    storage = params.storage
+  }
+
+  // Load persisted fields
+  if (persist) {
+    persist.forEach(key => {
+      const value = storage.get(`shared-data:${key}`)
```

Comment:
Perhaps we should use less generic storage key? Additional `vue` related prefix would rather be enough :)

Reply:
This is done on the devtools side, and thus the localStorage is already specific to the extension.